specs/WebMessageObjects.md

```diff
@@ -119,6 +138,64 @@ void WebView_WebMessageReceivedHandler(object sender, CoreWebView2WebMessageRece
 }
 ```
 
+## Post a FileSystemHandle to the web content
+Use this API to create a FileSystemHandle from the app and send it to the web content in WebView2.
+
+### Web content code
+```javascript
+chrome.webview.addEventListener("message", function (e) {
+    if (e.data.MyMessageType === "FileHandle") {
+        var fileHandle = e.additionalObjects[0];
+        if (fileHandle.kind === "file") {
+            // run file code
+        } else if (fileHandle.kind === "directory") {
+            // run directory code
+        }
+    }
+});
+```
+
+### App code
+```cpp
+wil::com_ptr<ICoreWebView2_21> webview21 =
+    m_webView2.try_query<ICoreWebView2_21>();
+wil::com_ptr<ICoreWebView2Environment> environment =
+    appWindow->GetWebViewEnvironment();
+wil::com_ptr<ICoreWebView2Environment14> environment14 = 
+    environment.try_query<ICoreWebView2Environment14>();
+wil::com_ptr<ICoreWebView2FileSystemHandle> fileHandle;
+CHECK_FAILURE(environment14->CreateWebFileSystemHandle(
+    L"C:\\Users\\<user>\\Documents\\file.txt",
+    COREWEBVIEW2_FILE_SYSTEM_HANDLE_PERMISSION_READWRITE,
+    COREWEBVIEW2_FILE_SYSTEM_HANDLE_KIND_FILE, &fileHandle));
+wil::com_ptr<ICoreWebView2ObjectCollection> webObjectCollection;
+IUnknown* webObjects[] = {fileHandle.get()};
+CHECK_FAILURE(environment14->CreateObjectCollection(
+    ARRAYSIZE(webObjects),
+    webObjects,
+    &webObjectCollection));
+// Check the source to ensure that we are sending the message to the correct target.
+wil::unique_cotaskmem_string source;
+CHECK_FAILURE(webview21->get_Source(&source));
+if (std::wstring(source.get()).rfind(L"https://www.example.com/", 0) == 0) {

```

Comment:
This seems like an awfully weird way of testing the source domain. It would reject `https://www.example.com/https://www.example.com/` which seems like it should be okay (still under the `www.example.com` domain).

Why not just `starts_with(L"https://www.example.com/")`?